presto-native-execution/presto_cpp/main/connectors/arrow_flight/ArrowFlightConnector.h

```diff
@@ -0,0 +1,194 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <arrow/flight/api.h>
```

Comment:
Let's use the PIMPL idiom and limit arrow headers to the cpp file.
See example usage here https://github.com/facebookincubator/velox/blob/main/velox/connectors/hive/storage_adapters/s3fs/S3FileSystem.h#L77

Reply:
I was able to move things around a bit and just forward declare the arrow classes to get rid of the arrow includes. They are still in a couple of the test headers, though. That's fine right?

Reply:
test headers are fine.